nuxeo-features/rest-api/nuxeo-rest-api-test/src/main/java/org/nuxeo/ecm/restapi/test/BaseAutomationTest.java

```diff
@@ -0,0 +1,247 @@
+        String schemas = null;
+        String acceptedType = MediaType.APPLICATION_JSON + ", */*";
+        String data = getAutomationBody(input, params, contextParams);
+        try (MultiPart mp = getAutomationMultiPart(input, data);
+                CloseableClientResponse response = getResponse(RequestType.POST, operationId, data, null, mp, schemas,
+                        headers, acceptedType); //
+                InputStream stream = response.getEntityInputStream()) {
+            int status = response.getStatus();
+            if (status == SC_NO_CONTENT) {
+                return null;
+            } else {
+                assertEquals(SC_OK, status);
+                return bodyReader.apply(stream);
+            }
+        } catch (IOException e) {
+            throw new NuxeoException(e);
+        }
+    }
+
+    protected String getOperationError(String operationId, Map<String, Object> params, int expectedStatus) {
+        return getOperationError(operationId, null, params, null, null, expectedStatus);
+    }
+
+    protected String getOperationError(String operationId, String input, Map<String, Object> params,
+            int expectedStatus) {
+        return getOperationError(operationId, input, params, null, null, expectedStatus);
+    }
+
+    protected String getOperationError(String operationId, String input, Map<String, Object> params,
+            Map<String, Object> contextParams, Map<String, String> headers, int expectedStatus) {
+        headers = headers == null ? new HashMap<>() : new HashMap<>(headers);
+        headers.put("Content-Type", MediaType.APPLICATION_JSON);
+        String schemas = null;
+        String acceptedType = MediaType.APPLICATION_JSON + ", */*";
+        String data = getAutomationBody(input, params, contextParams);
+        try (CloseableClientResponse response = getResponse(RequestType.POST, operationId, data, null, null, schemas,
+                headers, acceptedType); //
+                InputStream stream = response.getEntityInputStream()) {
+            assertEquals(expectedStatus, response.getStatus());
+            JsonNode node = mapper.readTree(stream);
+            if (node == null) {
+                return null;
+            }
+            JsonNode messageNode = node.get("message");
+            return messageNode == null ? null : messageNode.asText();
+        } catch (IOException e) {
+            throw new NuxeoException(e);
+        }
+    }
+
+    protected String getAutomationBody(Object input, Map<String, Object> params, Map<String, Object> contextParams) {
+        Map<String, Object> map = new HashMap<>();
+        if (input instanceof String) {
+            map.put("input", input);
+        }
+        if (params == null) {
+            params = Collections.emptyMap();
+        }
+        map.put("params", params);
+        if (contextParams == null) {
+            contextParams = Collections.emptyMap();
+        }
+        map.put("context", contextParams);
```

Comment:
What do you think of the following?
```java
map.put("context", requireNonNullElseGet(contextParams, Collections::emptyMap);
```

Reply:
In the end I use `params == null ? Map.of() : params`